src/shims/unix/fs.rs

```diff
@@ -1105,17 +1117,18 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 let ino = 0u64;
 
                 let file_type = this.file_type_to_d_type(dir_entry.file_type())?;
-
                 this.write_int_fields_named(
-                    &[
-                        ("d_ino", ino.into()),
-                        ("d_off", 0),
-                        ("d_reclen", size.into()),
-                        ("d_type", file_type.into()),
-                    ],
+                    &[("d_ino", ino.into()), ("d_off", 0), ("d_reclen", size.into())],
                     &this.ptr_to_mplace(entry, dirent64_layout),
                 )?;
 
+                if is_linux {
```

Comment:
I think ideally we'd say "if the d_type field exists, then write to it".

So I suggest you go to `helpers.rs` and find `fn project_field_named`. Then you add a new `fn try_project_field_named` that returns `InterpResult<'tcx, Option<P>>` so that when the field does not exist, it is `None`. And then you use that here.